bundles/org.openhab.binding.ism8/src/main/java/org/openhab/binding/ism8/server/Server.java

```diff
@@ -0,0 +1,271 @@
+            this.client = clientSocket;
+            logger.debug("Connection from Partner established {}", clientSocket.getRemoteSocketAddress());
+            if (listener != null) {
+                listener.connectionStatusChanged(ThingStatus.ONLINE);
+            }
+
+            this.startRetries = 0;
+            this.sendUpdateCommand();
+            while (!this.isInterrupted()) {
+                byte[] bytes = getBytesFromInputStream(clientSocket.getInputStream());
+                ArrayList<byte[]> packages = this.getPackages(bytes);
+
+                for (byte[] pack : packages) {
+                    logger.debug("Data received: {}", this.printBytes(pack));
+                    KnxNetFrame frame = KnxNetFrame.createKnxNetPackage(pack, pack.length);
+                    if (frame != null) {
+                        byte[] answer = frame.createFrameAnswer();
+                        if (answer.length > 0) {
+                            this.sendData(answer);
+                        }
+
+                        for (SetDatapointValueMessage message : frame.getValueMessages()) {
+                            logger.debug("Message received: {} {}", message.getId(),
+                                    this.printBytes(message.getData()));
+
+                            IDataPoint dataPoint = null;
+                            for (int k = 0; k < this.dataPoints.size(); k++) {
+                                IDataPoint dp = this.dataPoints.get(k);
+                                if (dp.getId() == message.getId()) {
+                                    dataPoint = dp;
+                                    break;
+                                }
+                            }
+
+                            if (dataPoint != null) {
+                                dataPoint.processData(message.getData());
+                                logger.debug("{} {}", dataPoint.getDescription(), dataPoint.getValueText());
+                                if (listener != null) {
+                                    listener.dataPointChanged(new DataPointChangedEvent(this, dataPoint));
+                                }
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+        } catch (IOException e) {
+            logger.warn("Error handle client data stream. {}", e.getMessage());
+            this.stopServer();
+        }
+    }
+
+    private void sendUpdateCommand() throws IOException {
+        byte[] data = new byte[] { (byte) 0x06, (byte) 0x20, (byte) 0xF0, (byte) 0x80, (byte) 0x00, (byte) 0x16,
+                (byte) 0x04, (byte) 0x00, (byte) 0x00, (byte) 0x00, (byte) 0xF0, (byte) 0xD0 };
+        this.sendData(data);
+    }
+
+    private byte[] getBytesFromInputStream(InputStream is) throws IOException {
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        byte[] buffer = new byte[0xFFFF];
```

Comment:
Basically - yes, it can be reduced. 
The ISM8 card sends packages upon any value change within the heating system, so basically rarely. 
Additionally, did I never detected any case where multiple data-points were combined within one package. Thus, the size of the package might be not bigger than 20-30 bytes. 
In the moment of establishing the connection I'm sending an active request to receive all data-points. After this command the ISM sends a series of packages (based on the physical configuration). Thus I'd not like to make this just 30 bytes as this might be too small in case multiple packages are within the buffer and other hardware than mine potentially have different data-points. Therefore I'd like to go for 255 (0xFF). Would this be fine from your point of view?